Publish quality attack flow diagrams.

curious-attempt-bunny opened the pull request:
Fixes #42 

## What Changed

1. Tidied up graphviz script output. Now it aligns better with the style of the manually created drawio diagrams.

### Before

![image](https://user-images.githubusercontent.com/51608/168922619-cf4bc617-cf6f-4415-9544-da594b480db5.png)

### After

![image](https://user-images.githubusercontent.com/51608/168922546-338e6ab1-d578-4398-9f84-34f563c73941.png)

## Limitations

1. Layout! (it's graphviz)
2. Text exceeding the fixed sized nodes.

mehaase commented on src/attack_flow/graphviz.py:
```diff
@@ -2,11 +2,39 @@ def convert(attack_flow):
     graph.append("}")
     return "\n".join(graph)
+
+def align_node_label(label: str, width=20) -> str:
```
Can you add a docstring to this method that explains what it does?

mehaase reviewed (changes requested):
Thanks for making this PR. The output looks great. I have one comment on the diff, and can you also please fix the [build errors](https://github.com/center-for-threat-informed-defense/attack-flow/runs/6489201008?check_suite_focus=true)? The build is failing because:

1. The graphviz test needs to be updated (run the test suite locally with `pytest --cov=src/ --cov-report term-missing`)
2. Some flake8 formatting issues, mainly long lines (>120 chars) in graphviz.py. I suggest unrolling some of the list comprehensions but I'll leave that up to you.

curious-attempt-bunny commented:
@mehaase I addressed the feedback. Although somehow there are now 116 Security Hotspots according to SonarCloud - but I'm unable to view the report.

I also added documentation for running the tests, added the flake checks to the makefile, duplicated the flake8 configuration so that it's picked up out of the box by the MS Visual Studio plugin, and

mehaase commented on src/attack_flow/graphviz.py:
```diff
@@ -1,12 +1,93 @@
+GRAPHVIZ_NODE_STYLE_COMMON = ''
```
Is this used anywhere? If not, please delete.

mehaase commented on src/attack_flow/graphviz.py:
```diff
@@ -1,12 +1,93 @@
+    for rel in attack_flow['relationships']:
+        if '#state' in rel.get("type", ""):
+            if rel["type"].endswith("#state-change"):
+                labelText = "provides"
```
Change to `label_text`.

curious-attempt-bunny commented:
@mehasse Cleaned up. Thanks for your time reviewing these changes.

mehaase commented:
Thank you for the contribution @curious-attempt-bunny! This will definitely be valuable to the community at large.
